drivers/serial/uart_mspm0g3xxx.c

```diff
@@ -115,9 +116,8 @@ static void uart_mspm0g3xxx_irq_tx_disable(const struct device *dev)
 
 static int uart_mspm0g3xxx_irq_tx_ready(const struct device *dev)
 {
-	const struct uart_mspm0g3xxx_config *config = dev->config;
-
-	return (DL_UART_getEnabledInterruptStatus(config->regs, UART_MSPM0_TX_INTERRUPTS)) ? 1 : 0;
+	struct uart_mspm0g3xxx_data *data = dev->data;
+	return data->pending_interrupt | UART_MSPM0_TX_INTERRUPTS ? 1 : 0;
```

Comment:
``` suggestion
	return data->pending_interrupt & UART_MSPM0_TX_INTERRUPTS;
```
Wouldn't this be cleaner?

Reply:
The documentation states that the method should return either 1 and 0.

The `|` is definitely wrong though. Should ofc. compare the pending interrupt instead

Reply:
Should be compared to the IIDX now instead. Used gdb to check the value of the pending interrupts and now the values are matching (symbols alignment as well) :+1: